plugin/src/main/kotlin/org/gradleweaver/plugins/jlink/JLinkTask.kt

```diff
@@ -4,132 +4,257 @@ import org.gradle.api.DefaultTask
 import org.gradle.internal.jvm.Jvm
+import org.gradle.internal.os.OperatingSystem
+import org.gradle.kotlin.dsl.listProperty
+import org.gradle.kotlin.dsl.property
 import java.io.ByteArrayOutputStream
+import java.io.File
+import java.lang.IllegalStateException
+import java.lang.module.ModuleFinder
+import java.lang.module.ModuleReference
 import java.nio.charset.Charset
-import javax.inject.Inject
+import java.nio.file.Paths
 
 // Note: must be open so Gradle can create a proxy subclass
-open class JLinkTask : DefaultTask {
+open class JLinkTask : DefaultTask() {
 
-    internal val options: JLinkOptions
+    private val objectFactory = project.objects
+    private val launcherGenerator = LauncherGenerator()
 
-    @Inject
-    constructor(options: JLinkOptions) {
-        this.options = options
-    }
+    /**
+     * The name of the generated image.
+     */
+    @get:Input
+    val imageName = objectFactory.property<String>()
+
+    /**
+     * Link service provider modules and their dependencies.
+     */
+    @get:Input
+    val bindServices = objectFactory.property<Boolean>()
+
+    /**
+     * Enable compression of resources.
+     */
+    @get:Input
+    val compressionLevel = objectFactory.property<CompressionLevel>()
+
+    /**
+     * Specifies the byte order of the generated image. The default value is the format of your system's architecture.
+     */
+    @get:Input
+    val endianness = objectFactory.property<Endianness>()
+
+    /**
+     * Suppresses a fatal error when signed modular JARs are linked in the runtime image.
+     * The signature-related files of the signed modular JARs are not copied to the runtime image.
+     */
+    @get:Input
+    val ignoreSigningInformation = objectFactory.property<Boolean>()
+
+    /**
+     * Excludes header files from the generated image.
+     */
+    @get:Input
+    val excludeHeaderFiles = objectFactory.property<Boolean>()
+
+    /**
+     * Excludes man pages from the generated image.
+     */
+    @get:Input
+    val excludeManPages = objectFactory.property<Boolean>()
+
+    /**
+     * Strips debug symbols from the generated image.
+     */
+    @get:Input
+    val stripDebug = objectFactory.property<Boolean>()
+
+    /**
+     * Optimize `Class.forName` calls to constant class loads.
+     */
+    @get:Input
+    val optimizeClassForName = objectFactory.property<Boolean>()
+
+    /**
+     * Options for generated launcher scripts. This is an optional property, and if it is not set, then no launcher
+     * script will be generated.
+     */
+    @get:Input
+    @get:Optional
+    val launcherOptions = objectFactory.property<JLinkLauncherOptions>()
 
-    @InputFile
-    var applicationJarLocation: RegularFileProperty = newInputFile()
+    @get:Input
+    val extraModules = objectFactory.listProperty<String>()
 
-    @OutputDirectory
-    var jlinkDir: DirectoryProperty = newOutputDirectory()
+    @get:InputFile
+    val applicationJarLocation: RegularFileProperty = newInputFile()
```

Comment:
Apparently, `newInputFile` is deprecated in Gradle 5.0.

Reply:
Welp. As long as it'll still work